cfpb/consumerfinance.gov
# New forms for FOIA/privacy requests

The FOIA and Privacy teams requested a fillable form to allow users to request records online, to fulfill a new requirement from OMB, due today. The two new forms, at `/privacy/records-access/` and `/privacy/disclosure-consent/`, send an email of their contents to the FOIA inbox, including any uploaded image files as attachments.


## Additions

- New pages at `/privacy/records-access/` and `/privacy/disclosure-consent`
- New django forms, `RecordsAcessForm` and `DisclosureConsentForm`, with tests
- Basic HTML email templates for sending the form contents as email
- Custom JS to add fields (mailing address) to the page only when they are required
- A basic file upload widget


## Changes

- Updates the Apache header `LIMIT_REQUEST_BODY` on the new privacy form pages so that users can upload files


## How to test this PR

1. Run this branch
2. In the Wagtail admin, add and enable the following flag: `PRIVACY_FORMS`
2. Visit `/privacy/records-access/`
3. Fill out the form
4. On submit, check the server log to see that it sent an HTML email containing the responses in the form.
5. Try submitting the form with invalid data (invalid email addresses, incorrect file types/sizes, omitting the mailing address info if 'please send my records by mail' is selected) and see that error messages appear correctly on the page


## Screenshots


## Notes and todos

- We're planning to add a few more tests.


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
